tests/core/pyspec/eth2spec/test/altair/transition/test_activation.py

```diff
@@ -0,0 +1,228 @@
+
+    validator_index = 0
+    signed_exits = prepare_signed_exits(spec, state, [validator_index])
+    operation_dict = {'voluntary_exits': signed_exits}
+
+    # irregular state transition to handle fork:
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch, operation_dict=operation_dict)
+    blocks = []
+    blocks.append(post_tag(block))
+
+    validator = state.validators[validator_index]
+    assert validator.exit_epoch < post_spec.FAR_FUTURE_EPOCH
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
+
+    yield "blocks", blocks
+    yield "post", state
+
+
+#
+# Activation
+#
+
+
+@fork_transition_test(PHASE0, ALTAIR, fork_epoch=2)
+def test_transition_with_non_empty_activation_queue(state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    Create some deposits before the transition
+    """
+    transition_until_fork(spec, state, fork_epoch)
+
+    _, queuing_indices = set_some_new_deposits(spec, state, rng=random.Random(5566))
+
+    assert spec.get_current_epoch(state) < fork_epoch
+    assert len(queuing_indices) > 0
+    for validator_index in queuing_indices:
+        assert not spec.is_active_validator(state.validators[validator_index], spec.get_current_epoch(state))
+
+    yield "pre", state
+
+    # irregular state transition to handle fork:
+    blocks = []
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch)
+    blocks.append(post_tag(block))
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
+
+    yield "blocks", blocks
+    yield "post", state
+
+
+@fork_transition_test(PHASE0, ALTAIR, fork_epoch=2)
+def test_transition_with_deposit_at_fork(state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    Create a deposit at the transition
+    """
+    transition_until_fork(spec, state, fork_epoch)
+
+    yield "pre", state
+
+    # create a new deposit
+    validator_index = len(state.validators)
+    amount = post_spec.MAX_EFFECTIVE_BALANCE
+    deposit = prepare_state_and_deposit(post_spec, state, validator_index, amount, signed=True)
+    deposit_old = prepare_state_and_deposit(spec, state, validator_index, amount, signed=True)
+    # sanity check: deposit operation is independent of spec fork versions
+    assert deposit_old == deposit
+    operation_dict = {'deposits': [deposit]}
+    # irregular state transition to handle fork:
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch, operation_dict=operation_dict)
+    blocks = []
+    blocks.append(post_tag(block))
+
+    assert not post_spec.is_active_validator(state.validators[validator_index], post_spec.get_current_epoch(state))
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
```

Comment:
But since `activation_eligibility_epoch` is 3, if I remove `transition_to_next_epoch_and_append_blocks `, I have to have to do one more epoch (slot_count=spec.SLOTS_PER_EPOCH * 3)` in `next_slots_with_attestations` to make ` state.finalized_checkpoint.epoch >= state.validators[validator_index].activation_eligibility_epoch`.

The status quo is a bit more efficient since `next_slots_with_attestations` is heavier.